Apps/AdminWebClient/src/Server/Services/CommunicationService.cs

```diff
@@ -74,25 +74,25 @@ public RequestResult<Communication> Add(Communication communication)
         /// <inheritdoc />
         public RequestResult<Communication> Update(Communication communication)
         {
-            if (ValidateDates(communication.EffectiveDateTime, communication.ExpiryDateTime))
```

Comment:
After reflecting on this, the date check is business logic and should be moved to the service.  The delegate can then handle either banner or email easily again and the DB will fail if needed.  This will make the code cleaner.

Reply:
+1